cni/pkg/nodeagent/server_linux.go

```diff
@@ -94,15 +97,24 @@ func initMeshDataplane(client kube.Client, args AmbientArgs) (*meshDataplane, er
 		return nil, fmt.Errorf("error creating traffic managers: %w", err)
 	}
 
-	if !useNftables {
-		// The nftables implementation will automatically flush any pre-existing chains when programming
-		// the rules, so we skip the DeleteHostRules for nftables backend.
-		hostTrafficManager.DeleteHostRules()
-	}
+	err = backoff.Retry(func() error {
+		if !useNftables {
+			// The nftables implementation will automatically flush any pre-existing chains when programming
+			// the rules, so we skip the DeleteHostRules for nftables backend.
+
+			// Not infallible, this can silently fail so we should retry it as well
+			hostTrafficManager.DeleteHostRules()
```

Comment:
Should we fail/initiate another retry at this stage if the host rules fail to delete? Not sure if it is worth preemptively failing or if the `innerErr` gives us enough indication of the failure reason.